lib/class-wp-rest-blocks-controller.php

```diff
@@ -16,6 +16,39 @@
  * @see WP_REST_Controller
  */
 class WP_REST_Blocks_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Checks if a block can be read.
+	 *
+	 * @since 2.1.0
+	 *
+	 * @param object $post Post object that backs the block.
+	 * @return bool Whether the block can be read.
+	 */
+	public function check_read_permission( $post ) {
+		// Ensure that the user is logged in and has the read_blocks capability.
+		$post_type = get_post_type_object( $post->post_type );
+		if ( ! current_user_can( $post_type->cap->read_post, $post->ID ) ) {
```

Comment:
Minor point, but you should be able to provide the `edit_post` meta capability here instead rather than looking up the primitive directly, since `map_meta_cap()` will be doing it for you later anyway. So then this line can be simply:

```php
if ( ! current_user_can( 'read_post', $post->ID ) ) {
```

Reply:
If you use `read_post` (not `read_block`) then `map_meta_cap()` should automatically map it over to `read_block` (or whatever the registered post type cap gets changed to).